src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -409,6 +450,37 @@ public override void ExecuteSiteRecoveryCmdlet()
                             ToDictionary(x => x.DiskId, x => x.DiskEncryptionSetId);
                     }
 
+                    if (this.RecoveryVmTag == null ||
+                      this.RecoveryVmTag.Count == 0)
+                    {
+                        recoveryVmTag = providerSpecificDetails.TargetVmTags;
+                    }
+
+                    if (this.RecoveryNicTag == null ||
+                       this.RecoveryNicTag.Count == 0)
+                    {
+                        recoveryNicTag = providerSpecificDetails.TargetNicTags;
+                    }
+
+                    if (this.DiskTag == null ||
+                      this.DiskTag.Count == 0)
+                    {
+                        diskTag = providerSpecificDetails.TargetManagedDiskTags;
+                    }
+
+                    if(useManagedDisk == Constants.False)
+                    {
+                        if(diskTag != null ||
+                            diskTag.Count > 0)
+                        {
```

Comment:
we are checking effectively (this.DiskTag == null ||
                      this.DiskTag.Count == 0)

twice
something like
if (this.DiskTag == null ||
                      this.DiskTag.Count == 0)
                    {
                        diskTag = providerSpecificDetails.TargetManagedDiskTags;
                    }
else if(useManagedDisk == Constants.False)
                    {throw new PSArgumentException(
                               string.Format(
                                   Resources.DiskTagCannotBeSet,
                                   diskTag,
                                   useManagedDisk));
                        }